### PR TITLE
fix(integrations): create Integrations menu & add domain config in domain store

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [x] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Integrations GNB menu

https://user-images.githubusercontent.com/83635051/219554010-dab35cc4-8ff2-4ebb-9c37-e8dc328614f1.mov

- Add `ExtraMenu` in domain store
- Add `Integrations` GNB menu
- Add `IntegrationSubMenu`

### CODE REVIEW

File: src/store/modules/domain/actions.ts [hunk 1]
```diff
@@ -55,3 +55,12 @@ export const setBillingEnabled: Action<DomainState, any> = async ({ commit, root
...
 };
+
+export const loadDomainConfig: Action<DomainState, any> = async ({ commit }) => {
+    try {
+        const { results } = await SpaceConnector.client.config.domainConfig.list({});
+        commit('setDomainConfig', results);
+    } catch (e) {
+        commit('setDomainConfig', undefined);
+    }
+};
```
Comment (WANZARGEN) [inline src/store/modules/domain/actions.ts:66]:
```suggestion
export const loadDomainConfig: Action<DomainState, any> = async ({ commit, state }) => {
   if (state.domainConfig) return;
    try { 
        const { results } = await SpaceConnector.client.config.domainConfig.list({});
        commit('setDomainConfig', results);
    } catch (e) {
        ErrorHandler.handle(e);
        commit('setDomainConfig', {});
    }
};
```

File: src/store/modules/domain/index.ts [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 
 const state: DomainState = {
     domainId: undefined,
     name: undefined,
     extendedAuthType: undefined,
     authOptions: undefined,
     billingEnabled: false,
+    domainConfig: {
+        [DOMAIN_CONFIG_TYPE.EXTRA_MENU]: undefined,
+    } as DomainConfigMap,
```
Comment (WANZARGEN) [inline src/store/modules/domain/index.ts:15]:
```suggestion
    domainConfig: undefined as DomainConfigMap|undefined,
```

File: src/store/modules/domain/mutations.ts [hunk 3]
```diff
@@ -10,3 +10,13 @@ export const setDomain = (state: DomainState, domainInfo: DomainState): void =>
...
+
+export const setDomainConfig = (state: DomainState, domainConfig: DomainConfigModel|undefined) => {
+    if (domainConfig?.length) {
+        const result = {} as DomainConfigMap;
+        domainConfig.forEach((config) => {
+            result[config.name] = config.data;
+        });
+        state.domainConfig = result;
+    }
+};
```
Comment (WANZARGEN) [inline src/store/modules/domain/mutations.ts:22]:
```suggestion
export const setDomainConfig = (state: DomainState, domainConfig: DomainConfig[]) => {
    if (domainConfig?.length) {
        const result = {} as DomainConfigMap;
        domainConfig.forEach((config) => {
            result[config.name] = config.data;
        });
        state.domainConfig = result;
    }
};
```

File: src/store/modules/domain/actions.ts [hunk 4]
```diff
@@ -55,3 +58,17 @@ export const setBillingEnabled: Action<DomainState, any> = async ({ commit, root
...
+        const { results } = await SpaceConnector.client.config.domainConfig.list({
+            name: DOMAIN_CONFIG_TYPE.EXTRA_MENU,
+        });
+        const result = results.find((res) => res.name === DOMAIN_CONFIG_TYPE.EXTRA_MENU);
+        commit('setDomainConfig', result?.data);
+    } catch (e) {
+        ErrorHandler.handleError(e);
+        commit('setDomainConfig', undefined);
+    }
+};
```
Comment (WANZARGEN) [inline src/store/modules/domain/actions.ts:74]:
```suggestion
export const loadExtraMenu: Action<DomainState, any> = async ({ commit, state }) => {
    if (state.extraMenu) return;
    try {
        const { results } = await SpaceConnector.client.config.domainConfig.list({
            name: DOMAIN_CONFIG_TYPE.EXTRA_MENU,
        });
        const result = results.find((res) => res.name === DOMAIN_CONFIG_TYPE.EXTRA_MENU);
        commit('setDomainConfig', result?.data ?? {});
    } catch (e) {
        ErrorHandler.handleError(e);
        commit('setDomainConfig',{});
    }
};
```

**Comment by piggggggggy** [comment]
Thank you for reviewing..........

**Review by WANZARGEN — APPROVED** [review]

**Pull request merged**